morpho-checkpoint-node/src/Morpho/RPC/Abstract.hs

```diff
@@ -66,13 +77,44 @@ instance ToJSON ev => ToJSON (RpcTrace ev i o) where
       ]
 
 instance Show e => MPretty (RpcTrace e i o) where
-  mpretty = viaShow
+  mpretty (RpcTrace GetLatestBlock request event) =
+    "GetLatestBlock(request:"
+      <+> mpretty request
+      <> ", event:"
+      <+> viaShow event
+      <> ")"
+  mpretty (RpcTrace PushCheckpoint request event) =
+    "PushCheckpoint(request:"
+      <+> mpretty request
+      <> ", event:"
+      <+> viaShow event
+      <> ")"
```

Comment:
This can have a nicer instance, currently it looks like

> GetLatestBlock(request: (CheckpointingInterval: 4, PowBlock(number: 236, hash: b45e6ad6)), event: RpcSuccess Nothing)

I think this would be much nicer e.g. like

> RPC call GetLatestBlock(interval: 4, parent: PowBlock(number: 236, hash: b45e6ad6)) returned: Nothing

For pushing:

> RPC call PushCheckpoint(Checkpoint(block: PowBlock(number: 236, hash: b45e6ad6))) returned: true

And for the `RpcStart` event:

> Starting RPC call GetLatestBlock(interval: 4, parent: PowBlock(number: 236, hash: b45e6ad6))


In general I think that all the `Trace` instances (like `SubscriptionTrace`, `RpcTrace`, etc.) should have a sentence-like structure. On the other hand, data structures like `MorphoBlock`, `Vote`, etc. should have a structural instance like `MorphoBlock(hash: ...)`. These structural instances can be used in the `Trace` instances.

Reply:
You can do a nested pattern match on the `event` to implement this

Reply:
@Infinisil If I understand correctly, I should only use the `RpcSuccess` data constructor for the event and leave out `RpcStart` and `RpcEvent` ?

Reply:
No, to have all of those be encoded as sentences, rather than structures. So for example:
```haskell
mpretty (RpcTrace GetLatestBlock (k, parent) RpcStart) = "Starting RPC call GetLatestBlock(interval:" <+> mpretty k <> ", parent:" <+> mpretty parent
mpretty (RpcTrace PushCheckpoint block (RpcSuccess result)) = "Success, RPC call PushCheckpoint(" <> mpretty block <> " returned" <+> mpretty result
```

And that for all combinations of the method and event